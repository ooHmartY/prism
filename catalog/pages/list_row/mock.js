import React from "react";

import colors from "../../../src/theme/colors";

import ParkingIcon from "../../../src/components/Icons/Parking";
import VenueIcon from "../../../src/components/Icons/Venue";
import TicketIcon from "../../../src/components/Icons/Ticket";
import VipIcon from "../../../src/components/Icons/Vip";
import ThumbnailCircleImage from "../../../src/components/Image/ThumbnailCircle";

export const listItems = [
  {
    rowId: "567",
    title: "Del Mar Fairgrounds",
    subTitle: "KABOO 3-Day Pass",
    dateTitle: "apr 23",
    dateSubTitle: "Thu, 8:00 PM",
    buttonText: "See Tickets",
    variant: "standard",
    onClick: () => {},
    url: "/"
  },
  {
    rowId: "569",
    title: "Del Mar Fairgrounds",
    subTitle: "KABOO 3-Day Pass",
    dateTitle: "apr 23",
    dateSubTitle: "Thu, 8:00 PM",
    buttonText: "See Tickets",
    variant: "standard",
    onClick: () => {},
    url: "/"
  },
  {
    rowId: "570",
    title: "Del Mar Fairgrounds",
    subTitle: "KABOO 3-Day Pass",
    dateTitle: "apr 23",
    dateSubTitle: "Thu, 8:00 PM",
    buttonText: "See Tickets",
    variant: "standard",
    onClick: () => {},
    url: "/"
  }
];

export const sections = [
  {
    title: "Ticket Options",
    key: "789",
    items: [
      {
        title: "VIP Packages",
        icon: <VipIcon size={26} color={colors.blackPearl} />,
        onItemClick: () => {}
      },
      {
        title: "Parking Pass Available",
        icon: <ParkingIcon size={26} color={colors.blackPearl} />
      }
    ],
    totalSections: 4
  },

  {
    title: "Venue Info",
    key: "456",
    items: [
      {
        title: "United Center",
        subTitle: "Chicago, IL",
        icon: <VenueIcon size={26} color={colors.blackPearl} />,
        onItemClick: () => {},
        url: "/"
      }
    ],
    totalSections: 4
  },

  {
    title: "Lineup",
    key: "123",
    items: [
      {
        title:
          "The Roots tour of San Diego california hdjhj wjdwjd wjdhwejkdh wdjewj",
        icon: (
          <ThumbnailCircleImage
            size={40}
            src="https://placekitten.com/g/512/288"
            alt="circleImage"
          />
        ),
        onItemClick: () => {}
      },
      {
        title: "T.I",
        icon: (
          <ThumbnailCircleImage
            size={40}
            src="https://placekitten.com/g/512/288"
            alt="circleImage"
          />
        ),
        onItemClick: () => {}
      }
    ],
    totalSections: 4
  },
  {
    title: "More",
    key: "1011",
    items: [
      {
        title: "Sell Tickets",
        icon: <TicketIcon size={26} color={colors.blackPearl} />,
        onItemClick: () => {}
      }
    ],
    totalSections: 4
  }
];

export const items = [
  {
    title: "United Center",
    subTitle: "Chicago, IL",
    icon: <VenueIcon size={26} color={colors.blackPearl} />,
    onItemClick: () => {},
    url: "/"
  }
];
