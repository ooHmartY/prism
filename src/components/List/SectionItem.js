import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import spacing from "../../theme/spacing";
import { SecondaryText, Link } from "../Text";
import { ItemContainerConsumer } from "./Context";
import { LinkTitle } from "../Text/Link";

const ItemContainer = styled.div.attrs({
  className: "section-item"
})`
  display: flex;
  flex-direction: row;
  align-items: center;
  cursor: ${props => (props.onItemClick ? "pointer" : "text")};

  &:not(:last-of-type) {
    margin-bottom: ${spacing.moderate};
  }
`;

const ItemContainerLink = ItemContainer.withComponent(Link);

const IconContainer = styled.div`
  padding-right: ${spacing.cozy};
`;

/* stylelint-disable */
const MultiLineLink = LinkTitle.withComponent("div").extend`
  display: -webkit-box;
  overflow: hidden;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
`;
/* stylelint-enable */

const handleItemClick = (children, value, event, onItemClick) => {
  if (children && value) {
    value.renderIntoPortal({ children, contentType: "desktop" });
  }

  if (onItemClick) {
    onItemClick(event);
  }
};

const SectionItem = ({
  item: { icon, title, subTitle, onItemClick, url },
  children,
  ...props
}) => {
  const Container = url ? ItemContainerLink : ItemContainer;
  const Title = onItemClick || url ? MultiLineLink : SecondaryText;

  return (
    <ItemContainerConsumer>
      {value => (
        <Container
          key={title}
          role="link"
          aria-label="Section Item"
          onClick={event =>
            handleItemClick(children, value, event, onItemClick)
          }
          onItemClick={onItemClick}
          href={url}
          {...props}
        >
          {icon && <IconContainer>{icon}</IconContainer>}
          <div>
            <Title>{title}</Title>
            {subTitle && <SecondaryText>{subTitle}</SecondaryText>}
          </div>
        </Container>
      )}
    </ItemContainerConsumer>
  );
};

SectionItem.defaultProps = {
  children: null
};

SectionItem.propTypes = {
  item: PropTypes.shape({
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string,
    url: PropTypes.string,
    icon: PropTypes.node,
    onItemClick: PropTypes.func
  }).isRequired,
  children: PropTypes.node
};

export default SectionItem;
