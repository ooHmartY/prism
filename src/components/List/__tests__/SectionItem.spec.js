import React from "react";
import { render, Simulate } from "react-testing-library";

import { sections } from "../../../../catalog/pages/list_row/mock";
import SectionItem from "../SectionItem";
import ListContainer from "../Container";

describe("<SectionItem />", () => {
  beforeEach(() => {
    const modalRoot = global.document.createElement("div");
    modalRoot.setAttribute("id", "modal-root");
    modalRoot.setAttribute("data-testid", "modal-root");
    const body = global.document.querySelector("body");
    body.appendChild(modalRoot);
  });

  afterEach(() => {
    const body = global.document.querySelector("body");
    const modalRoot = global.document.getElementById("modal-root");
    body.removeChild(modalRoot);
  });

  it('renders SectionItem without "url" correctly', () => {
    const { container } = render(
      <SectionItem
        item={sections[0].items[0]}
        key={sections[0].items[0].title}
      />
    );
    Simulate.click(container.querySelector(".section-item"));
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders SectionItem with "url" correctly', () => {
    const { container } = render(
      <SectionItem
        item={{ ...sections[0].items[0], url: "https://ticketmaster.com/" }}
        key={sections[0].items[0].title}
      />
    );
    Simulate.click(container.querySelector(".section-item"));
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders SectionItem with children correctly", () => {
    const { container } = render(
      <ListContainer onRowCollapse={() => {}} onModalClose={() => {}}>
        <SectionItem
          item={sections[0].items[0]}
          key={sections[0].items[0].title}
        >
          <div>Asia</div>
          <div>Africa</div>
          <div>North America</div>
        </SectionItem>
      </ListContainer>
    );
    Simulate.click(container.querySelector(".section-item"));
    expect(container.firstChild).toMatchSnapshot();
  });
});
