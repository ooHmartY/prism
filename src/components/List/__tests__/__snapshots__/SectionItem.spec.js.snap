// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionItem /> renders SectionItem with "url" correctly 1`] = `
<a
  aria-label="Section Item"
  class="sc-gZMcBi section-item nFpiP sc-gzVnrw bppGNV"
  color=""
  href="https://ticketmaster.com/"
  rel="_self"
  role="link"
  target=""
>
  <div
    class="sc-gqjmRU iYZLDL"
  >
    <svg
      height="26"
      viewBox="0 0 26 26"
      width="26"
    >
      <g
        fill="none"
        fill-rule="evenodd"
        transform="translate(1 .85)"
      >
        <circle
          cx="12"
          cy="12.15"
          fill="#FAFAFA"
          r="12.5"
          stroke="#E0E0E0"
        />
        <path
          d="M19.203 9.797c0 .302-.099.565-.297.789-.198.224-.44.357-.726.398-.073.302-.187.783-.34 1.442a54.47 54.47 0 0 0-.867 4.691 16.86 16.86 0 0 0-.176 2.281.398.398 0 0 1-.29.387.52.52 0 0 1-.198.004.337.337 0 0 1-.09-.031 3.182 3.182 0 0 0-.707-.227 11.24 11.24 0 0 0-1.067-.176c-.406-.049-.851-.087-1.336-.113-.484-.026-.987-.039-1.507-.039-.526 0-1.032.013-1.516.04-.484.025-.928.063-1.332.112-.404.05-.76.109-1.066.176a3.18 3.18 0 0 0-.711.227.396.396 0 0 1-.579-.36c0-.671-.057-1.432-.171-2.28a47.28 47.28 0 0 0-.403-2.49 54.47 54.47 0 0 0-.46-2.202c-.154-.66-.268-1.14-.34-1.442a1.158 1.158 0 0 1-.731-.398A1.162 1.162 0 0 1 4 9.796c0-.327.117-.608.352-.843a1.16 1.16 0 0 1 .851-.351c.328 0 .61.117.844.351.234.235.351.516.351.844 0 .224-.054.427-.164.61-.109.182-.252.325-.43.429a74.312 74.312 0 0 1 .86 3.922c.11.578.206 1.161.29 1.75.234-.084.51-.155.827-.215.318-.06.656-.11 1.016-.152-.01-.532-.044-1.17-.102-1.914A89.46 89.46 0 0 0 8.496 12a201.513 201.513 0 0 0-.379-3.438 1.155 1.155 0 0 1-.656-.417 1.165 1.165 0 0 1-.258-.747c0-.328.117-.609.352-.843a1.15 1.15 0 0 1 .843-.352c.334 0 .618.117.852.352.234.234.352.515.352.843 0 .24-.064.457-.192.653a1.17 1.17 0 0 1-.496.433c.037.308.09.773.16 1.395s.142 1.3.215 2.031c.073.732.14 1.47.2 2.215.06.745.094 1.393.105 1.945.27-.02.541-.036.812-.047.271-.01.537-.018.797-.023V6.133a1.205 1.205 0 0 1-.578-.434A1.152 1.152 0 0 1 10.398 5c0-.333.118-.617.352-.852a1.16 1.16 0 0 1 .852-.351c.328 0 .609.117.843.351.235.235.352.519.352.852 0 .26-.074.493-.223.7-.148.205-.34.35-.574.433V16c.26.005.526.013.797.023.27.01.539.026.805.047.015-.552.052-1.2.109-1.945.057-.745.122-1.483.195-2.215.073-.732.146-1.409.219-2.031l.164-1.395a1.17 1.17 0 0 1-.496-.433 1.167 1.167 0 0 1-.191-.653c0-.328.117-.609.351-.843a1.15 1.15 0 0 1 .844-.352c.333 0 .617.117.851.352.235.234.352.515.352.843 0 .282-.087.53-.262.747a1.171 1.171 0 0 1-.652.418c-.037.302-.091.765-.164 1.39a124.006 124.006 0 0 0-.418 4.273c-.06.745-.093 1.383-.098 1.915.354.041.69.092 1.008.152.318.06.596.131.836.215.083-.589.18-1.172.29-1.75.108-.578.217-1.121.323-1.63.107-.507.209-.96.305-1.358l.223-.934a1.213 1.213 0 0 1-.43-.43 1.162 1.162 0 0 1-.164-.61c0-.327.117-.608.351-.843A1.16 1.16 0 0 1 18 8.602c.333 0 .617.117.852.351.234.235.351.516.351.844zM14.797 7c-.11 0-.203.04-.281.117a.384.384 0 0 0-.118.281c0 .11.04.204.118.282a.384.384 0 0 0 .28.117c.115 0 .212-.04.29-.117a.384.384 0 0 0 .117-.282c0-.109-.039-.203-.117-.28a.393.393 0 0 0-.29-.118zM8.398 7c-.109 0-.203.04-.28.117A.384.384 0 0 0 8 7.398c0 .11.04.204.117.282a.384.384 0 0 0 .281.117c.11 0 .204-.04.282-.117a.384.384 0 0 0 .117-.282c0-.109-.04-.203-.117-.28A.384.384 0 0 0 8.398 7zm3.204-2.398c-.11 0-.204.039-.282.117a.384.384 0 0 0-.117.281c0 .11.04.203.117.281a.384.384 0 0 0 .282.117c.109 0 .203-.039.28-.117A.384.384 0 0 0 12 5c0-.11-.04-.203-.117-.281a.384.384 0 0 0-.281-.117zM4.797 9.797c0 .114.039.21.117.289a.393.393 0 0 0 .29.117c.109 0 .202-.039.28-.117a.393.393 0 0 0 .118-.29c0-.109-.04-.202-.118-.28a.384.384 0 0 0-.28-.118c-.115 0-.212.04-.29.118a.384.384 0 0 0-.117.28zm6.805 8.601c.37 0 .756.008 1.16.024.403.016.8.04 1.187.074.388.034.76.078 1.114.133.354.055.671.124.953.207a19.058 19.058 0 0 1 .125-1.516 4.056 4.056 0 0 0-.735-.215c-.307-.065-.659-.12-1.054-.164a26.146 26.146 0 0 0-2.75-.145c-.506 0-.993.014-1.461.04-.47.026-.901.061-1.297.105-.396.045-.748.1-1.055.164-.307.066-.55.137-.726.215l.074.778c.023.252.04.498.05.738.277-.083.592-.152.946-.207.354-.055.726-.1 1.117-.133.39-.034.786-.058 1.188-.074.4-.016.789-.024 1.164-.024zM18 10.203c.11 0 .203-.039.281-.117a.393.393 0 0 0 .117-.29c0-.109-.039-.202-.117-.28A.384.384 0 0 0 18 9.398c-.11 0-.203.04-.281.118a.384.384 0 0 0-.117.28c0 .115.039.212.117.29a.384.384 0 0 0 .281.117z"
          fill="#1F262D"
        />
      </g>
    </svg>
  </div>
  <div>
    <div
      class="sc-jTzLTM bvEXjR"
    >
      VIP Packages
    </div>
  </div>
</a>
`;

exports[`<SectionItem /> renders SectionItem with children correctly 1`] = `
<div
  class="sc-kEYyzF cIuKYE"
>
  <div
    aria-label="Section Item"
    class="sc-iwsKbI section-item kEeBdL"
    role="link"
  >
    <div
      class="sc-gqjmRU iYZLDL"
    >
      <svg
        height="26"
        viewBox="0 0 26 26"
        width="26"
      >
        <g
          fill="none"
          fill-rule="evenodd"
          transform="translate(1 .85)"
        >
          <circle
            cx="12"
            cy="12.15"
            fill="#FAFAFA"
            r="12.5"
            stroke="#E0E0E0"
          />
          <path
            d="M19.203 9.797c0 .302-.099.565-.297.789-.198.224-.44.357-.726.398-.073.302-.187.783-.34 1.442a54.47 54.47 0 0 0-.867 4.691 16.86 16.86 0 0 0-.176 2.281.398.398 0 0 1-.29.387.52.52 0 0 1-.198.004.337.337 0 0 1-.09-.031 3.182 3.182 0 0 0-.707-.227 11.24 11.24 0 0 0-1.067-.176c-.406-.049-.851-.087-1.336-.113-.484-.026-.987-.039-1.507-.039-.526 0-1.032.013-1.516.04-.484.025-.928.063-1.332.112-.404.05-.76.109-1.066.176a3.18 3.18 0 0 0-.711.227.396.396 0 0 1-.579-.36c0-.671-.057-1.432-.171-2.28a47.28 47.28 0 0 0-.403-2.49 54.47 54.47 0 0 0-.46-2.202c-.154-.66-.268-1.14-.34-1.442a1.158 1.158 0 0 1-.731-.398A1.162 1.162 0 0 1 4 9.796c0-.327.117-.608.352-.843a1.16 1.16 0 0 1 .851-.351c.328 0 .61.117.844.351.234.235.351.516.351.844 0 .224-.054.427-.164.61-.109.182-.252.325-.43.429a74.312 74.312 0 0 1 .86 3.922c.11.578.206 1.161.29 1.75.234-.084.51-.155.827-.215.318-.06.656-.11 1.016-.152-.01-.532-.044-1.17-.102-1.914A89.46 89.46 0 0 0 8.496 12a201.513 201.513 0 0 0-.379-3.438 1.155 1.155 0 0 1-.656-.417 1.165 1.165 0 0 1-.258-.747c0-.328.117-.609.352-.843a1.15 1.15 0 0 1 .843-.352c.334 0 .618.117.852.352.234.234.352.515.352.843 0 .24-.064.457-.192.653a1.17 1.17 0 0 1-.496.433c.037.308.09.773.16 1.395s.142 1.3.215 2.031c.073.732.14 1.47.2 2.215.06.745.094 1.393.105 1.945.27-.02.541-.036.812-.047.271-.01.537-.018.797-.023V6.133a1.205 1.205 0 0 1-.578-.434A1.152 1.152 0 0 1 10.398 5c0-.333.118-.617.352-.852a1.16 1.16 0 0 1 .852-.351c.328 0 .609.117.843.351.235.235.352.519.352.852 0 .26-.074.493-.223.7-.148.205-.34.35-.574.433V16c.26.005.526.013.797.023.27.01.539.026.805.047.015-.552.052-1.2.109-1.945.057-.745.122-1.483.195-2.215.073-.732.146-1.409.219-2.031l.164-1.395a1.17 1.17 0 0 1-.496-.433 1.167 1.167 0 0 1-.191-.653c0-.328.117-.609.351-.843a1.15 1.15 0 0 1 .844-.352c.333 0 .617.117.851.352.235.234.352.515.352.843 0 .282-.087.53-.262.747a1.171 1.171 0 0 1-.652.418c-.037.302-.091.765-.164 1.39a124.006 124.006 0 0 0-.418 4.273c-.06.745-.093 1.383-.098 1.915.354.041.69.092 1.008.152.318.06.596.131.836.215.083-.589.18-1.172.29-1.75.108-.578.217-1.121.323-1.63.107-.507.209-.96.305-1.358l.223-.934a1.213 1.213 0 0 1-.43-.43 1.162 1.162 0 0 1-.164-.61c0-.327.117-.608.351-.843A1.16 1.16 0 0 1 18 8.602c.333 0 .617.117.852.351.234.235.351.516.351.844zM14.797 7c-.11 0-.203.04-.281.117a.384.384 0 0 0-.118.281c0 .11.04.204.118.282a.384.384 0 0 0 .28.117c.115 0 .212-.04.29-.117a.384.384 0 0 0 .117-.282c0-.109-.039-.203-.117-.28a.393.393 0 0 0-.29-.118zM8.398 7c-.109 0-.203.04-.28.117A.384.384 0 0 0 8 7.398c0 .11.04.204.117.282a.384.384 0 0 0 .281.117c.11 0 .204-.04.282-.117a.384.384 0 0 0 .117-.282c0-.109-.04-.203-.117-.28A.384.384 0 0 0 8.398 7zm3.204-2.398c-.11 0-.204.039-.282.117a.384.384 0 0 0-.117.281c0 .11.04.203.117.281a.384.384 0 0 0 .282.117c.109 0 .203-.039.28-.117A.384.384 0 0 0 12 5c0-.11-.04-.203-.117-.281a.384.384 0 0 0-.281-.117zM4.797 9.797c0 .114.039.21.117.289a.393.393 0 0 0 .29.117c.109 0 .202-.039.28-.117a.393.393 0 0 0 .118-.29c0-.109-.04-.202-.118-.28a.384.384 0 0 0-.28-.118c-.115 0-.212.04-.29.118a.384.384 0 0 0-.117.28zm6.805 8.601c.37 0 .756.008 1.16.024.403.016.8.04 1.187.074.388.034.76.078 1.114.133.354.055.671.124.953.207a19.058 19.058 0 0 1 .125-1.516 4.056 4.056 0 0 0-.735-.215c-.307-.065-.659-.12-1.054-.164a26.146 26.146 0 0 0-2.75-.145c-.506 0-.993.014-1.461.04-.47.026-.901.061-1.297.105-.396.045-.748.1-1.055.164-.307.066-.55.137-.726.215l.074.778c.023.252.04.498.05.738.277-.083.592-.152.946-.207.354-.055.726-.1 1.117-.133.39-.034.786-.058 1.188-.074.4-.016.789-.024 1.164-.024zM18 10.203c.11 0 .203-.039.281-.117a.393.393 0 0 0 .117-.29c0-.109-.039-.202-.117-.28A.384.384 0 0 0 18 9.398c-.11 0-.203.04-.281.118a.384.384 0 0 0-.117.28c0 .115.039.212.117.29a.384.384 0 0 0 .281.117z"
            fill="#1F262D"
          />
        </g>
      </svg>
    </div>
    <div>
      <div
        class="sc-jTzLTM bvEXjR"
      >
        VIP Packages
      </div>
    </div>
  </div>
</div>
`;

exports[`<SectionItem /> renders SectionItem without "url" correctly 1`] = `
<div
  aria-label="Section Item"
  class="sc-iwsKbI section-item kEeBdL"
  role="link"
>
  <div
    class="sc-gqjmRU iYZLDL"
  >
    <svg
      height="26"
      viewBox="0 0 26 26"
      width="26"
    >
      <g
        fill="none"
        fill-rule="evenodd"
        transform="translate(1 .85)"
      >
        <circle
          cx="12"
          cy="12.15"
          fill="#FAFAFA"
          r="12.5"
          stroke="#E0E0E0"
        />
        <path
          d="M19.203 9.797c0 .302-.099.565-.297.789-.198.224-.44.357-.726.398-.073.302-.187.783-.34 1.442a54.47 54.47 0 0 0-.867 4.691 16.86 16.86 0 0 0-.176 2.281.398.398 0 0 1-.29.387.52.52 0 0 1-.198.004.337.337 0 0 1-.09-.031 3.182 3.182 0 0 0-.707-.227 11.24 11.24 0 0 0-1.067-.176c-.406-.049-.851-.087-1.336-.113-.484-.026-.987-.039-1.507-.039-.526 0-1.032.013-1.516.04-.484.025-.928.063-1.332.112-.404.05-.76.109-1.066.176a3.18 3.18 0 0 0-.711.227.396.396 0 0 1-.579-.36c0-.671-.057-1.432-.171-2.28a47.28 47.28 0 0 0-.403-2.49 54.47 54.47 0 0 0-.46-2.202c-.154-.66-.268-1.14-.34-1.442a1.158 1.158 0 0 1-.731-.398A1.162 1.162 0 0 1 4 9.796c0-.327.117-.608.352-.843a1.16 1.16 0 0 1 .851-.351c.328 0 .61.117.844.351.234.235.351.516.351.844 0 .224-.054.427-.164.61-.109.182-.252.325-.43.429a74.312 74.312 0 0 1 .86 3.922c.11.578.206 1.161.29 1.75.234-.084.51-.155.827-.215.318-.06.656-.11 1.016-.152-.01-.532-.044-1.17-.102-1.914A89.46 89.46 0 0 0 8.496 12a201.513 201.513 0 0 0-.379-3.438 1.155 1.155 0 0 1-.656-.417 1.165 1.165 0 0 1-.258-.747c0-.328.117-.609.352-.843a1.15 1.15 0 0 1 .843-.352c.334 0 .618.117.852.352.234.234.352.515.352.843 0 .24-.064.457-.192.653a1.17 1.17 0 0 1-.496.433c.037.308.09.773.16 1.395s.142 1.3.215 2.031c.073.732.14 1.47.2 2.215.06.745.094 1.393.105 1.945.27-.02.541-.036.812-.047.271-.01.537-.018.797-.023V6.133a1.205 1.205 0 0 1-.578-.434A1.152 1.152 0 0 1 10.398 5c0-.333.118-.617.352-.852a1.16 1.16 0 0 1 .852-.351c.328 0 .609.117.843.351.235.235.352.519.352.852 0 .26-.074.493-.223.7-.148.205-.34.35-.574.433V16c.26.005.526.013.797.023.27.01.539.026.805.047.015-.552.052-1.2.109-1.945.057-.745.122-1.483.195-2.215.073-.732.146-1.409.219-2.031l.164-1.395a1.17 1.17 0 0 1-.496-.433 1.167 1.167 0 0 1-.191-.653c0-.328.117-.609.351-.843a1.15 1.15 0 0 1 .844-.352c.333 0 .617.117.851.352.235.234.352.515.352.843 0 .282-.087.53-.262.747a1.171 1.171 0 0 1-.652.418c-.037.302-.091.765-.164 1.39a124.006 124.006 0 0 0-.418 4.273c-.06.745-.093 1.383-.098 1.915.354.041.69.092 1.008.152.318.06.596.131.836.215.083-.589.18-1.172.29-1.75.108-.578.217-1.121.323-1.63.107-.507.209-.96.305-1.358l.223-.934a1.213 1.213 0 0 1-.43-.43 1.162 1.162 0 0 1-.164-.61c0-.327.117-.608.351-.843A1.16 1.16 0 0 1 18 8.602c.333 0 .617.117.852.351.234.235.351.516.351.844zM14.797 7c-.11 0-.203.04-.281.117a.384.384 0 0 0-.118.281c0 .11.04.204.118.282a.384.384 0 0 0 .28.117c.115 0 .212-.04.29-.117a.384.384 0 0 0 .117-.282c0-.109-.039-.203-.117-.28a.393.393 0 0 0-.29-.118zM8.398 7c-.109 0-.203.04-.28.117A.384.384 0 0 0 8 7.398c0 .11.04.204.117.282a.384.384 0 0 0 .281.117c.11 0 .204-.04.282-.117a.384.384 0 0 0 .117-.282c0-.109-.04-.203-.117-.28A.384.384 0 0 0 8.398 7zm3.204-2.398c-.11 0-.204.039-.282.117a.384.384 0 0 0-.117.281c0 .11.04.203.117.281a.384.384 0 0 0 .282.117c.109 0 .203-.039.28-.117A.384.384 0 0 0 12 5c0-.11-.04-.203-.117-.281a.384.384 0 0 0-.281-.117zM4.797 9.797c0 .114.039.21.117.289a.393.393 0 0 0 .29.117c.109 0 .202-.039.28-.117a.393.393 0 0 0 .118-.29c0-.109-.04-.202-.118-.28a.384.384 0 0 0-.28-.118c-.115 0-.212.04-.29.118a.384.384 0 0 0-.117.28zm6.805 8.601c.37 0 .756.008 1.16.024.403.016.8.04 1.187.074.388.034.76.078 1.114.133.354.055.671.124.953.207a19.058 19.058 0 0 1 .125-1.516 4.056 4.056 0 0 0-.735-.215c-.307-.065-.659-.12-1.054-.164a26.146 26.146 0 0 0-2.75-.145c-.506 0-.993.014-1.461.04-.47.026-.901.061-1.297.105-.396.045-.748.1-1.055.164-.307.066-.55.137-.726.215l.074.778c.023.252.04.498.05.738.277-.083.592-.152.946-.207.354-.055.726-.1 1.117-.133.39-.034.786-.058 1.188-.074.4-.016.789-.024 1.164-.024zM18 10.203c.11 0 .203-.039.281-.117a.393.393 0 0 0 .117-.29c0-.109-.039-.202-.117-.28A.384.384 0 0 0 18 9.398c-.11 0-.203.04-.281.118a.384.384 0 0 0-.117.28c0 .115.039.212.117.29a.384.384 0 0 0 .281.117z"
          fill="#1F262D"
        />
      </g>
    </svg>
  </div>
  <div>
    <div
      class="sc-jTzLTM bvEXjR"
    >
      VIP Packages
    </div>
  </div>
</div>
`;
